saw-core/prelude/Prelude.sawcore

```diff
@@ -309,31 +309,37 @@ piCong1 r x y eq =
 --------------------------------------------------------------------------------
 -- Booleans
 
--- Boolean is a primitive type, because it is handled specially by some of the
--- back-ends (e.g., the SAT/SMT solvers)
```

Comment:
This now-removed comment claims that there is some special handling for `Bool` in the backends. I assume that whatever special treatment exists still works with these new changes?